src/cli/node/with_node.rs

```diff
@@ -145,8 +153,12 @@ pub fn exec_cmd(command: Command, mut config: Config) -> Result<(), failure::Err
             let write_to_path = match (write, write_to) {
                 // Don't write
                 (false, None) => None,
-                // Write to default storage path
-                (true, None) => Some(config.storage.db_path),
+                // Write to the same folder where the config file is located
+                // Fail if using default configuration (not sourced from a file in the filesystem)
+                (true, None) => config_path
+                    .expect("Cannot guess a file system path for writing the master key file. Please specify a valid path right after the `--write` flag.")
+                    .parent()
+                    .map(Path::to_path_buf),
```

Comment:
Same issue, if the config path is "/", `.parent()` will return `None` and the master key will be silently not-written.

Reply:
Oops, I merged cos' it was approved, but then saw your comment @tmpolaczyk. I'll do a little PR adding an `expect` after `parent()`. Sorry for pulling the trigger so quickly!